src/tools.rs

```diff
@@ -229,13 +229,17 @@ pub fn guess_language<P: AsRef<Path>>(buf: &[u8], path: P) -> (Option<LANG>, Str
     }
 }
 
+/// Replaces \n and \r ending characters with a single generic \n
 pub(crate) fn remove_blank_lines(data: &mut Vec<u8>) {
-    let count_trailing = data.iter().rev().take_while(|&c| *c == b'\n').count();
+    let count_trailing = data
+        .iter()
+        .rev()
+        .take_while(|&c| (*c == b'\n' || *c == b'\r'))
```

Comment:
This is going to count Windows newlines twice.
On Linux you would have `Ciao\nCiao\n\n`, and so count would be 2.
On Windows you would have `Ciao\r\nCiao\r\n\r\n`, and so count would be 4.

Reply:
Forget it, the code after is removing the characters and not the lines.